### DISCUSSION TITLE
Documentation lacking.

### DISCUSSION THREAD

**Discussion by alexmitylite** [WordPress/gutenberg#50556] [Q&A]
I'm trying to get a RichText to display an unordered list but the documentation doesn't seem to tell me what all my options are.

I'm at this [link ](https://developer.wordpress.org/block-editor/reference-guides/richtext/) reading about RichText. But the only thing I can see is an example like this (allowedFormats={ [ 'core/bold', 'core/italic' ] }).

Where can I go to see a full list of all the allowedFormats? I got chat gpt to give me this list.

'core/bold': Makes the selected text bold.
'core/italic': Makes the selected text italic.
'core/text-color': Changes the color of the selected text.
'core/link': Makes the selected text a hyperlink.
'core/strikethrough': Strikes through the selected text.
'core/underline': Underlines the selected text.
'core/code': Formats the selected text as code.
'core/subscript': Formats the selected text as subscript.
'core/superscript': Formats the selected text as superscript.
'core/keyboard': Formats the selected text as keyboard input.
'core/inline-code': Formats the selected text as inline code.
'core/list/unordered': Formats the selected text as an unordered list.
'core/list/ordered': Formats the selected text as an ordered list.

But I cannot for the life of me figure out where it got that info from so I can't tell if it's accurate.

core/list/unordered didn't work for me which is why I'm trying to find documentaion on what is supported. I have this problem constantly with Wordpress blocks, I have no clue what my options are in general.

**Comment by alexmitylite** [comment]
I'm also looking at this [link ](https://github.com/WordPress/gutenberg/blob/HEAD/packages/block-editor/src/components/rich-text/README.md) here. Unfortunately it gives me the same example.

allowedFormats={ [ 'core/bold', 'core/italic' ] }

But it doesn't list all possible options.

**Comment by devfle** [comment]
You get a full list of all allowed Format-Types with the following command:
```javascript
wp.data.select('core/rich-text').getFormatTypes()
```

if you need some inspiration, how to create a list block, then you can check out the Gutenberg list-item block:  https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/list-item/edit.js#LL79C14-L79C14

if you are searching for the options to set the list ordered or unordered you can find some inspiration in the list block: https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/list/edit.js#L145

**Comment by marvinpoo** [comment]
Sadly there is no documentation for that, though, you can check the core formats in the Wordpress Gutenberg Repository under `packages/format-library/src/default-formats.js` 

[https://github.com/WordPress/gutenberg/blob/caae2893eb3d2c7ffe168fb5e668a2809fd932ea/packages/format-library/src/default-formats.js](https://github.com/WordPress/gutenberg/blob/caae2893eb3d2c7ffe168fb5e668a2809fd932ea/packages/format-library/src/default-formats.js)

These are all the possible core formats. Though be aware; Plugins can potentially override these.
